### ISSUE TITLE
No longer working for Vue component

### ISSUE THREAD

**Issue by nick-stanley** [SublimeLinter/SublimeLinter-jshint#113]
Was working fine for [Vue.js single file components](https://vuejs.org/v2/guide/single-file-components.html) up until recently.

![jshint for Vue single file components in Sublime](https://i.imgur.com/kmpVD6f.png)

I'm not sure if this is the right place for this. If not, sorry, and I'm happy to be pointed in the right direction.

**Comment by kaste** [comment]
Here, the vue syntax reports the script/js content wrong.

**Comment by kevin611003** [comment]
+1
I have to add this `excludes": ["*/*.vue"]`
 `
"linters": {
    "jshint": {
        "disable": false,
        "args": [],
        "excludes": ["*/*.vue"]
    }
 }
`

**Comment by braver** [comment]
Pro-tip: use eslint with the vue plugin. 

With some work you can get the selector to work most of the times, but the syntax is just completely broken. They can't expect any plugin to work with the vue syntax in its current state.

**Comment by Meekohi** [comment]
So is there no way to make this work with `jshint`? `eslint` is too agro for our codebase... but we need support for single-file `*.vue` components.

**Comment by braver** [comment]
Sure, there is. As I said, the root cause is:

> the syntax is just completely broken

If the vue syntax cleans up its act to properly delineate the different embedded syntaxes, both eslint and jshint can be made to deal with it without plugins. Someone just needs to put a little effort into the vue syntax. It's not hard, and probably not even more than a few hours work. If you care about this combo for your team, perhaps you can give it a try. A syntax is basically just a bunch of regexes, not rocket science.

**Comment by braver** [comment]
This tells us what’s what: https://packagecontrol.io/packages/Vue%20Syntax%20Highlight.

**Comment by jgthms** [comment]
Just wanted to give my 2 cents as I ran into the same issue. Here's a quick example:

![Screenshot 2019-10-10 at 09 09 57](https://user-images.githubusercontent.com/1254808/66551574-a8dc7d00-eb3f-11e9-92eb-2ca82cae5f3f.png)

I have the Vue Syntax Highlight plugin installed as well, which does highlight my code correctly. But the linter doesn't understand what's in the `<script>` element. So if I fix according to what Sublime tells me, it doesn't match what the actually linting config wants:

![Screenshot 2019-10-10 at 09 09 57](https://user-images.githubusercontent.com/1254808/66551683-d9bcb200-eb3f-11e9-8ef5-5d51f57aeeb4.png)

Anyway, I did what @kevin611003 suggested and added this to my `SublimeLinter.sublime-settings` file:

```json
{
  "linters": { "eslint": { "disable": false, "args": [], "excludes": ["*/*.vue"] } }
}
```

Hope this helps anyone.

**Comment by jgthms** [comment]
Just wanted to give my 2 cents as I ran into the same issue. Here's a quick example:

![Screenshot 2019-10-10 at 09 09 57](https://user-images.githubusercontent.com/1254808/66551574-a8dc7d00-eb3f-11e9-92eb-2ca82cae5f3f.png)

I have the Vue Syntax Highlight plugin installed as well, which does highlight my code correctly. But the linter doesn't understand what's in the `<script>` element. So if I fix according to what Sublime tells me, it doesn't match what the actually linting config wants:

![Artboard](https://user-images.githubusercontent.com/1254808/66551821-1e484d80-eb40-11e9-88a3-99b5fbe609a7.jpg)

Anyway, I did what @kevin611003 suggested and added this to my `SublimeLinter.sublime-settings` file:

```json
{
  "linters": { "eslint": { "disable": false, "args": [], "excludes": ["*/*.vue"] } }
}
```

Hope this helps anyone.
